client/java/src/main/java/com/linecorp/centraldogma/client/TransformingWatcher.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.centraldogma.client;
+
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.centraldogma.common.Revision;
+
+public class TransformingWatcher<T, U> implements Watcher<U> {
+
+    protected final Watcher<T> parent;
```

Comment:
How about renaming `parent` into `delegate` or `origin`?
I don't think we have to put `parent` child relationship here. 
This `TransformingWatcher` just map or transform original watcher. 😀
What do you think @trustin @minwoox ?

Reply:
How about:

- Adding `Watcher.mappedWatcher()` which is a static factory method.
- Renaming `newChild()` into `mappedWatcher()` and making it an overloaded static factory method (so that a user can always look for `mappedWatcher()` when he or she needs to transform.)

...?

Reply:
> Renaming newChild() into mappedWatcher()

I like the name, thanks for fast feedback. 😊

Reply:
I think the relationship is meaningful because the returned object is "forked" from the current watcher.
- Closing a forked watcher does nothing
- if transformer == identity, we still return a forked watcher that does not close its parent

WDYT?

Reply:
> if transformer == identity, we still return a forked watcher that does not close its parent

SGTM.